### DISCUSSION TITLE
Pycalphad error message on phases with sublattices having vacancies on both sublattices

### DISCUSSION THREAD

**Discussion by zliang1990** [pycalphad/pycalphad#502] [Q&A] [answered]
Hi Richard (@richardotis),

My name is Zhi Liang, Materials Design Engineer in QuesTek. When I recently attempted to run the attached V-O description (GitHub does not support .TDB so I zipped it) with Pycalphad, I noticed that the following error message will pop up if I only choose oxygen (and vacancy) as elements.

Upon debugging, it seems to do with the HALITE phase, whose sublattice model is V,V+2,V+3,VA : O-2,VA. When only choosing oxygen, first sublattice only allows VA while second sublattice allows O-2 and VA, which seemingly triggers the following error message. This issue can be a blocker if in a multicomponent system, certain phase only has sublattice model A,VA : O-2,VA, but the element selection only has element B and oxygen, and this error message will pop up as well (Sorry that I cannot provide our multicomponent database since it is proprietary).

Could you please help to see how we can resolve this?

Thanks!

Best,
Zhi

![image](https://github.com/pycalphad/pycalphad/assets/84593259/fce335ce-9037-4f6c-8a72-e112d55c2a3e)
[15YANG_V-O description.zip](https://github.com/pycalphad/pycalphad/files/13666266/15YANG_V-O.description.zip)

**Comment by bocklund** [comment]
What's happening here is that when only vacancies and oxygen are entered as elements PyCalphad is treating CORUNDUM and HALITE as active phases (because there are active species in each sublattice).

For both phases, it's impossible to charge balance given these components (pure vacancy endmembers don't count, such as in the case for HALITE) and that gives array size errors because there are no feasible `points`. I'm creating an issue for this, since it's a bug. We should probably prevent phases that cannot charge balance from being included in the active set of phases.

**Reply by zliang1990** [reply]
Thanks very much for identifying and helping!

**Reply by richardotis** [reply, marked as answer]
This is fixed in #432 and will be fixed in the next release version after that PR is merged.

**Reply by zliang1990** [reply]
Hello, I'm aware that the fix branch related to this discussion is still being thoroughly tested. I'm wondering if it is possible to separate this fix into a separate PR so we can test/use it as soon as possible? Thanks! I tagged my colleague Sam Sorkin (@sorkins) in this since he could provide some further background if needed.

**Reply by richardotis** [reply]
The fix for this issue is tied up with other new additions in #432 and cannot easily be separately backported. As I told Sam earlier this week, the core development team is meeting in-person next month to try to get the PR merged into mainline.
